doc/developer-notes.md

```diff
@@ -460,7 +460,10 @@ other input.
   safely continue even if the assumption is violated. In debug builds it
   behaves like `Assert`/`assert` to notify developers and testers about
   nonfatal errors. In production it doesn't warn or log anything, though the
-  expression is always evaluated.
+  expression is always evaluated. However, if the compiler can prove that a
+  statement inside `Assume` is side-effect-free, it may optimize the call away,
```

Comment:
According to https://www.reddit.com/r/cpp_questions/comments/plg8ij/expression_vs_statement/
> Statements do not have a return type

So by definition, unless they're empty, they cannot be `side-effect-free` - if my interpretation is correct.
```suggestion
  expression is always evaluated. However, if the compiler can prove that
  an expression inside `Assume` is side-effect-free, it may optimize the call away,
```

Reply:
Fixed!